### PR TITLE
controls the verbosity of killing tasks

### PR DESCRIPTION
**Pull request by AugustoPeres**
This pull request addresses: https://github.com/inductiva/tasks/issues/109

## Contents

When killing several the output was too verbose:

```
augustoperes@Augustos-Air inductiva % inductiva tasks kill --all                                              
You are about to kill ALL tasks.
Are you sure you want to proceed (y/[N])? y
Successfully sent kill request for task g7ic65a4iy15ifnecn7p1g6r8.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id g7ic65a4iy15ifnecn7p1g6r8'.
Successfully sent kill request for task k8g5z9q8b0pieeojkua4zmue3.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id k8g5z9q8b0pieeojkua4zmue3'.
Successfully sent kill request for task 0k0oamoo0q84ugzfhney2k0zd.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id 0k0oamoo0q84ugzfhney2k0zd'.
Successfully sent kill request for task dnzznpsmq3c99arfnfrmtb4zi.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id dnzznpsmq3c99arfnfrmtb4zi'.
Successfully sent kill request for task k04dbctvvqc7d5cri64a2o651.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id k04dbctvvqc7d5cri64a2o651'.
Successfully sent kill request for task 1697107060966412965.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id 1697107060966412965'.
Successfully sent kill request for task 0jnzq8tc88r584pz6uzmdfydt.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id 0jnzq8tc88r584pz6uzmdfydt'.
Successfully sent kill request for task 1695999258293624322.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id 1695999258293624322'.
```

This pull request adds a new verbosity level to `task.kill()` that displays a shorter message:

```
augustoperes@Augustos-Air inductiva % inductiva tasks  kill --all
You are about to kill ALL tasks.
Are you sure you want to proceed (y/[N])? y
Successfully sent kill request for m04agmt0w2mn5awoyibvn14qi.
Successfully sent kill request for bou80b3fgg0k4g6o7r1s93z0b.
Successfully sent kill request for 5h269fxwjgz3fgzr0xgnn1sf3.
Successfully sent kill request for cnm8kjta57d58a7jo20mdsbwk.
Successfully sent kill request for 5nr2df41uiepwcv32po8nm03s.
Successfully sent kill request for nn76e1fas9dpkyxiazf551fra.
Successfully sent kill request for 6a3kaqmio0th5f3t3jngokjwq.
Successfully sent kill request for tzhorjdhvgur9fmncc29hjbvz.
Successfully sent kill request for q1mo3i64a0oix69srrauze1mi.
```

## Behaviour

Instead of adding this new behaviour only when the user kills all tasks I added it when the user kill more than one task. Please give you opinion on when we should trigger lower verbosity:

```
augustoperes@Augustos-Air inductiva % inductiva tasks kill 9m2pv45cir3vho5k8swxfywcb
You are about to kill the following tasks:
  - 9m2pv45cir3vho5k8swxfywcb
Are you sure you want to proceed (y/[N])? y
Successfully sent kill request for task 9m2pv45cir3vho5k8swxfywcb.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id 9m2pv45cir3vho5k8swxfywcb'.

augustoperes@Augustos-Air inductiva % inductiva tasks kill q9dca5g4yo0m0qzywa5ktvuzr b7zcxip8ur1aod1gc8b8zq22k
You are about to kill the following tasks:
  - b7zcxip8ur1aod1gc8b8zq22k
  - q9dca5g4yo0m0qzywa5ktvuzr
Are you sure you want to proceed (y/[N])? y
Successfully sent kill request for b7zcxip8ur1aod1gc8b8zq22k.
Successfully sent kill request for q9dca5g4yo0m0qzywa5ktvuzr.
```

### CODE REVIEW

**Comment by IvanPombo** [comment]
Can't you even be less verbose with:

```
Successfully sent kill request for:
m04agmt0w2mn5awoyibvn14qi.
bou80b3fgg0k4g6o7r1s93z0b.
5h269fxwjgz3fgzr0xgnn1sf3.
cnm8kjta57d58a7jo20mdsbwk.
5nr2df41uiepwcv32po8nm03s.
nn76e1fas9dpkyxiazf551fra.
 6a3kaqmio0th5f3t3jngokjwq.
tzhorjdhvgur9fmncc29hjbvz.
q1mo3i64a0oix69srrauze1mi.
```

**Comment by AugustoPeres** [comment]
> Can't you even be less verbose with:
> 
> ```
> Successfully sent kill request for:
> m04agmt0w2mn5awoyibvn14qi.
> bou80b3fgg0k4g6o7r1s93z0b.
> 5h269fxwjgz3fgzr0xgnn1sf3.
> cnm8kjta57d58a7jo20mdsbwk.
> 5nr2df41uiepwcv32po8nm03s.
> nn76e1fas9dpkyxiazf551fra.
>  6a3kaqmio0th5f3t3jngokjwq.
> tzhorjdhvgur9fmncc29hjbvz.
> q1mo3i64a0oix69srrauze1mi.
> ```

mmm, not sure we should. I think that the kill message despite the level of verbosity, should always contain information about the fact that the task was killed. Not just the ID.

for example:

```python
task.kill(..., verbosity_level=0)
```

should probably always print "Successfully sent kill request to ...." instead of just "<task_id>"

Additionally, that way we would have to first send the request to all tasks and, only after all of them having received successfully the kill status, can we print the ones that succeeded. The same for the ones that failed

**Comment by Pbarbosa92** [comment]
Something like this

img1

**Comment by AugustoPeres** [comment]
> Something like this
> 
> img1

Thanks for the comment. Will implement that

File: inductiva/tasks/task.py [hunk 1]
```diff
@@ -259,10 +261,9 @@ def _check_if_pending_kill(
...
-    def kill(
-            self,
-            wait_timeout: Optional[Union[float,
-                                         int]] = None) -> Union[bool, None]:
+    def kill(self,
+             wait_timeout: Optional[Union[float, int]] = None,
+             verbosity_level: int = 1) -> Union[bool, None]:
         """Request a task to be killed.
         
         This method requests that the current task is remotely killed.
```
Comment (Pbarbosa92) [inline inductiva/tasks/task.py:269]: Don't forget to update the doctoring

File: inductiva/tasks/task.py [hunk 2]
```diff
@@ -53,6 +53,8 @@ class Task:
     KILLABLE_STATUSES = {models.TaskStatusCode.SUBMITTED
                         }.union(RUNNING_STATUSES)
 
+    TASK_KILL_VERBOSITY_LEVELS = [0, 1, 2]
```
Comment (Pbarbosa92) [inline inductiva/tasks/task.py:56]:
```suggestion
    KILL_VERBOSITY_LEVELS = [0, 1, 2]
```

Comment (Pbarbosa92) [reply inductiva/tasks/task.py:56]: You are already on the Task class.

**Review by Pbarbosa92 — APPROVED** [review]

**Review by IvanPombo — CHANGES_REQUESTED** [review]
While this works for this application it is not a scalable approach, since we would have to change this on every function. 

I think there is a mismatch with the purpose of this PR and the verbosity/logging of things. 

The verbosity on the general Python code should be controlled by the logging system, since there users can just select the warning/errors/info/debug messages. Not by an argument on the function. That is not general behavior and in a script users would need to had it everywhere. That is why the natural way would be through the logging or an environment variable. 

The verbosity on the CLI is more immediate and closer to what we are trying to achieve. In general, it is selected with a flag `-v`, or -vv for extra verbosity to get the different verbosity levels. But as is now, the prints from the CLI are dependent on the python code, which I think is not the appropriate way, since they should probably be decoupled. 

I am not completely sure on the implementation, but I am neither comfortable with this approach, since it seems to integrate/hide a recurring problem. So I think, we should discuss ir further and see if it is crucial to proceed for this version.

**Review by IvanPombo — APPROVED** [review]
Going to free this to be merged, since we already discussed the next steps to tackle the concerns I raised.

**Comment by AugustoPeres** [comment]
@IvanPombo and @Pbarbosa92 Calling for QA:

The steps are simple. launch tasks and try with both `kill --all` and `kill task_id` to observe differences in behaviour. You should do the same from `python`:

```python
task.kill(...,verbosity_level=...)
```

To test several verbosity levels

**Comment by Pbarbosa92** [comment]
**QA** 🟢 

---killing tasks on the CLI---
kill --all
Captura de ecrã 2024-02-07, às 10 12 21
kill 3 tasks
Captura de ecrã 2024-02-07, às 10 13 42
---killing tasks on a script---

```
--- kill verbosity 0

--- kill verbosity 1
Successfully sent kill request for hq4f5cxd77magk2peu5mv56p5.
--- kill verbosity 2
Successfully sent kill request for task hh3yt4ctvc8eig59kciz59331.
> Consider tracking the task status by calling task.get_status()
> or via the CLI using 'inductiva tasks -id hh3yt4ctvc8eig59kciz59331'.
```

LGTM!

**Pull request merged**